search-commons/src/main/java/no/unit/nva/search/constants/ApplicationConstants.java

```diff
@@ -188,4 +190,18 @@ private static NestedAggregationBuilder generateLabelsAggregation(String object)
     private static String jsonPath(String... args) {
         return String.join(JSON_PATH_DELIMITER, args);
     }
+
+    private static FilterAggregationBuilder generateHasFileAggregation() {
+
+        FilterAggregationBuilder typeFilterAggregation = AggregationBuilders.filter("associatedArtifacts.type",
+                                                                                    QueryBuilders.termQuery("associatedArtifacts.type.keyword", "PublishedFile"));
+
+        FilterAggregationBuilder adminAgreementFilterAggregation = AggregationBuilders.filter("associatedArtifacts"
+                                                                                + ".administrativeAgreement",
+                                                                                QueryBuilders.termQuery("associatedArtifacts.administrativeAgreement", false));
+
```

Comment:
Use jsonPath and constants instead. This applied to  all lines in this method

QueryBuilders.termQuery(jsonPath(ASSOCIATED_ARTIFACTS,ADMINISTRATIVE_AGREEMENT, false));